src/svlog/codegen.rs

```diff
@@ -949,13 +1065,24 @@ where
                     panic!("no value emitted for {:?}", id);
                 }
             };
-            let inputs = prok.inputs.iter().map(lookup_value).collect();
-            let outputs = prok.outputs.iter().map(lookup_value).collect();
+            let inputs: Vec<_> = prok.inputs.iter().map(lookup_value).collect();
+            let outputs: Vec<_> = prok.outputs.iter().map(lookup_value).collect();
             let ext_unit = self.builder.add_extern(
                 self.into.unit(prok.unit).name().clone(),
                 self.into.unit(prok.unit).sig().clone(),
             );
-            self.builder.ins().inst(ext_unit, inputs, outputs);
+            self.builder.ins().inst(
+                ext_unit,
+                inputs.iter().map(|x| x.0).collect(),
+                outputs.iter().map(|x| x.0).collect(),
+            );
+            circt::llhd::InstanceOp::new(
+                self.mlir_builder,
+                "",
```

Comment:
Yeah good point, there are a few of these empty names around currently. For the instance here it's a mistake, because the instance definitely has a name in SV, so we can just use that. But in general it might make sense to have the LLHD dialect use optional names for signals and instances on the CIRCT side, so we don't have to encode a lack of name through `""`.

Reply:
I agree with optional names for signals. Not sure about instances. Let's say we have a nameless instance of an entity containing a named signal. The simulator would use the names of the instances in the hierarchy to identify this signal for each instance in the trace. Should we just skip these signals in the trace then? Could this potentially confuse a user who searches for this signal in the trace, but cannot find it because there was a nameless instance somewhere up in the hierarchy?

Reply:
Yeah you're right. Unnamed signals can just be skipped, but instance probably have to be named for the user not to get confused. Or we also just skip those in the trace.